src/opm/parser/eclipse/EclipseState/SimulationConfig/ThresholdPressure.cpp

```diff
@@ -89,13 +89,20 @@ namespace Opm {
 
         //Option is set and keyword is found
         if( m_active && thpresKeyword ) {
-            if( !hasEqlnumKeyword )
+#ifdef ENABLE_3DPROPS_TESTING
+            if (!fp.has<int>("EQLNUM"))
+                throw std::runtime_error("Error when internalizing THPRES: EQLNUM keyword not found in deck");
+
+            const auto& eqlnum = fp.get<int>("EQLNUM");
+#else
+            if( !eclipseProperties.hasDeckIntGridProperty("EQLNUM"))
                 throw std::runtime_error("Error when internalizing THPRES: EQLNUM keyword not found in deck");
 
-            //Find max of eqlnum
             const auto& eqlnumKeyword = eclipseProperties.getIntGridProperty( "EQLNUM" );
             const auto& eqlnum = eqlnumKeyword.getData();
-            maxEqlnum = *std::max_element(eqlnum.begin(), eqlnum.end());
+#endif
+            //Find max of eqlnum
+            int maxEqlnum = *std::max_element(eqlnum.begin(), eqlnum.end());
```

Comment:
This dereferences `eqlnum.end()` if `eqlnum` is empty.  Are we guaranteed that that does not happen?

Reply:
The `eqlnum` data is default initialized to guarantee that safely.